### PR TITLE
[ch354] Move to tab layout for Match Summary

### PR DESCRIPTION
**Pull request by Jarrydcrawford**

### CODE REVIEW

File: src/components/TabbedLayout.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+interface TabbedLayoutState {
+  activeTab: string;
+}
+
+export const TabbedLayout: React.FunctionComponent<TabbedLayoutProps> = ({
+  children,
+  defaultTab,
+}) => {
+  const [state, setState] = React.useState<TabbedLayoutState>({
+    activeTab: defaultTab || children[0].props.label || '',
```
Comment (harrygr) [inline src/components/TabbedLayout.tsx]: You shouldn't access the props of child components. This goes against the main concepts of react - component heirachy and it's an unsafe operation - who's to say the child has a `label` prop? Or that `children[0]` is defined/not null?

File: src/components/TabbedLayout.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  });
+
+  return (
+    <>
+      <TabList>
+        {React.Children.map(children, (child: React.ReactElement) => {
```
Comment (harrygr) [inline src/components/TabbedLayout.tsx]: This is also nasty. You lose all type safety and these apis aren't really encouraged anymore.

Comment (Jarrydcrawford) [reply src/components/TabbedLayout.tsx]: They take generics but it's truly an awful api. Getting rid of them.

File: src/components/TabbedLayout.tsx (hunk 1 above)
Comment (Jarrydcrawford) [reply src/components/TabbedLayout.tsx]: I was going to add type-safety but if I'm doing routes then it's pointless now.

File: src/pages/Match/index.tsx [hunk 3]
```diff
@@ -49,46 +56,97 @@ export const MatchIndex = (props: Props) => {
...
-              pageTitle="Match"
+              pageTitle="Match Summary"
               key="match"
-              header={<MatchInfo match={match} />}
+              header={
+                <Versus
+                  awayTeam={match.opponent}
+                  awayTeamScore={0}
+                  homeTeam={match.team.name}
+                  homeTeamScore={0}
```
Comment (harrygr) [inline src/pages/Match/index.tsx]: perhaps if the score is not yet defined we just show a `-`

**Pull request merged**